asserts/ifacedecls_test.go

```diff
@@ -268,3 +273,239 @@ foo: [{p: "zzz"}, {p: "/foo/y"}]
+	checkAttrs(c, rule.AllowAutoConnection.SlotAttributes, "sa5")
+	c.Assert(rule.DenyAutoConnection, NotNil)
+	checkAttrs(c, rule.DenyAutoConnection.PlugAttributes, "pa6")
+	checkAttrs(c, rule.DenyAutoConnection.SlotAttributes, "sa6")
+}
+
+func (s *plugSlotRulesSuite) TestCompilePlugRuleShortcutTrue(c *C) {
+	rule, err := asserts.CompilePlugRule("iface", "true")
+	c.Assert(err, IsNil)
+
+	c.Check(rule.Interface, Equals, "iface")
+	// install subrules
+	c.Assert(rule.AllowInstallation, NotNil)
+	c.Check(rule.AllowInstallation.PlugAttributes, Equals, asserts.AlwaysMatchAttributes)
+	c.Assert(rule.DenyInstallation, NotNil)
+	c.Check(rule.DenyInstallation.PlugAttributes, Equals, asserts.NeverMatchAttributes)
+	// connection subrules
+	checkBoolPlugConnConstraints(c, rule.AllowConnection, true)
+	checkBoolPlugConnConstraints(c, rule.DenyConnection, false)
+	// auto-connection subrules
+	checkBoolPlugConnConstraints(c, rule.AllowAutoConnection, true)
+	checkBoolPlugConnConstraints(c, rule.DenyAutoConnection, false)
+}
+
+func (s *plugSlotRulesSuite) TestCompilePlugRuleShortcutFalse(c *C) {
+	rule, err := asserts.CompilePlugRule("iface", "false")
+	c.Assert(err, IsNil)
+
+	// install subrules
+	c.Assert(rule.AllowInstallation, NotNil)
+	c.Check(rule.AllowInstallation.PlugAttributes, Equals, asserts.NeverMatchAttributes)
+	c.Assert(rule.DenyInstallation, NotNil)
+	c.Check(rule.DenyInstallation.PlugAttributes, Equals, asserts.AlwaysMatchAttributes)
+	// connection subrules
+	checkBoolPlugConnConstraints(c, rule.AllowConnection, false)
+	checkBoolPlugConnConstraints(c, rule.DenyConnection, true)
+	// auto-connection subrules
+	checkBoolPlugConnConstraints(c, rule.AllowAutoConnection, false)
+	checkBoolPlugConnConstraints(c, rule.DenyAutoConnection, true)
+}
+
+func (s *plugSlotRulesSuite) TestCompilePlugRuleDefaults(c *C) {
+	rule, err := asserts.CompilePlugRule("iface", map[string]interface{}{
+		"deny-auto-connection": "true",
+	})
+	c.Assert(err, IsNil)
+
+	// everything follows the defaults...
+
+	// install subrules
+	c.Assert(rule.AllowInstallation, NotNil)
+	c.Check(rule.AllowInstallation.PlugAttributes, Equals, asserts.AlwaysMatchAttributes)
+	c.Assert(rule.DenyInstallation, NotNil)
+	c.Check(rule.DenyInstallation.PlugAttributes, Equals, asserts.NeverMatchAttributes)
+	// connection subrules
+	checkBoolPlugConnConstraints(c, rule.AllowConnection, true)
+	checkBoolPlugConnConstraints(c, rule.DenyConnection, false)
+	// auto-connection subrules
+	checkBoolPlugConnConstraints(c, rule.AllowAutoConnection, true)
+	// ... but deny-auto-connection is on
+	checkBoolPlugConnConstraints(c, rule.DenyAutoConnection, true)
+}
+
+func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsIDConstraints(c *C) {
+	rule, err := asserts.CompilePlugRule("iface", map[string]interface{}{
+		"allow-connection": map[string]interface{}{
+			"slot-snap-id":      []interface{}{"snapid1", "snapid2"},
+			"slot-publisher-id": []interface{}{"canonical", "$same"},
```

Comment:
Instead of $same, it might be more useful to have something like $plug_account_id and $slot_account_id, so it's both clear and we can reuse the concept in other fields if necessary, without having to use different language on each.

It won't make a difference here, though. Just brainstorming.